babynames.py

```diff
@@ -61,20 +81,29 @@ def create_parser():
     return parser
 
 
+def summary(text, filename):
+    with open(filename, "w+") as file:
```

Comment:
I would not use "w+" mode here ... isn't that for writing/appending?  I think each time you write the summary file, you want to overwrite whatever may have been previously there instead of having the .summary file grow larger each time